### PR TITLE
Assign roles without using the submit button on user create page

### PR DESCRIPTION
**Pull request by grotlue**
## Description
On the user creation page one can add and remove roles. 
This PR will remove the "Assign role" button from this view and assign roles when they are selected instead.

## Motivation and Context
When creating user the actual user creation does only happen when submitting the whole form, so there is no need for an additional button and thus step when adding or removing roles.

Closes https://github.com/Graylog2/graylog2-server/issues/9933

### Note:
To unify the behaviour we should also adjust the role selector on the team creation page.
Additionally the user selector works the same way, this should also be adjusted.

## How Has This Been Tested?
**Environment:** Local

- Created a new user, selected and deselected roles
- Checked the new user after creation

✅ Roles were saved accordingly

### Before:
Monosnap screencast 2022-08-23 12-48-31

### After:
Monosnap screencast 2022-08-23 12-51-42

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by mnin — APPROVED** [review]
LGTM

File: graylog2-web-interface/src/components/permissions/RolesSelector.tsx [hunk 1]
```diff
@@ -59,19 +60,19 @@ const _options = (roles, assignedRolesIds, identifier) => roles
...
 
-const _assignRole = (selectedRoleName, roles, onSubmit, setSelectedRoleName, setIsSubmitting, setError) => {
-  if (!selectedRoleName) {
+const _assignRole = (selectedRoleNames, roles, onSubmit, setselectedRoleNames, setIsSubmitting, setError) => {
+  if (!selectedRoleNames) {
     return;
   }
 
-  const selectedRoleNames = selectedRoleName.split(',');
+  const selectedRoleNamess = selectedRoleNames.split(',');
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/permissions/RolesSelector.tsx:68]: We should call this `selectedRolesNameArray` for example to make it more readable.

**Review by ousmaneo — CHANGES_REQUESTED** [review]
Thank you for the changes this looks good to me. I added a small comment inline

**Review by ousmaneo — APPROVED** [review]
LGTM!

**Pull request merged**